### ISSUE TITLE
Using Google Maps JS API Loader (or alternate solution)

### ISSUE THREAD

**Issue by eric-schmidt** [fawmi/vue-google-maps#108]
First off, this library rocks, and has been a HUGE help so far -- so thanks for putting it together! Unfortunately, I'm hitting a little glitch when trying to use the Maps JavaScript API on multiple components on the same page. See below:

### Is your feature request related to a problem? Please describe.
I have a parent component that leverages this library in a child component; however, the parent component also needs to add the Maps Javascript API in order to take advantage of the Distance Matrix API. Unfortunately, when adding the JS API in the parent component, I am met with errors due to the child component trying to add the same JS.

[According to Google](https://cloud.google.com/blog/products/maps-platform/loading-google-maps-platform-javascript-modern-web-applications), it is recommended to use the [Google Maps JavaScript API Loader](https://www.npmjs.com/package/@googlemaps/js-api-loader), which appears to handle any errors that can arise from multiple components adding the same script.

Is it possible for this library to leverage Google's loader, or is there perhaps another route I can go in order to use the API on a parent component that does not have any context for whether the child component has/has not already added the API?

### Describe the solution you'd like.
Refactor this library to use the Google Maps JavaScript API Loader, or provide an alternate route for being able to use the JS API in multiple, unrelated components.

**Comment by eric-schmidt** [comment]
As a workaround, and using https://github.com/fawmi/vue-google-maps/issues/88 as an example, I realized that you can bubble [template refs](https://vuejs.org/guide/essentials/template-refs.html) up multiple components. Since I needed the `google` object from two components down, I was just able to add a `MapReference` ref to my child component as well the `GMapMap` component on that same child compoent. This then let me access the `google` object with something like this:

```
  mounted () {
    this.$refs.MapReference.$refs.MapReference.$mapPromise.then(() => {
      this.google = window.google;
    });
  },
```

**Comment by rjp2525** [comment]
I'm using this package in a SPA that has some other areas that utilize an address autocomplete component, so the Google Maps API is already linked at the top of my root page.

However this package injects another script tag no matter what, which causes an error to be thrown in the console:
```
You have included the Google Maps API multiple times on this page. This may cause unexpected errors.
```

It'd be ideal if there was an additional check in that logic or to disable the automatic injection entirely for cases like this.

**Comment by gegerino** [comment]
Is there a solution for the:
`You have included the Google Maps API multiple times on this page. This may cause unexpected errors.`
Error?

**Comment by jaimemoransantos** [comment]
I'm having the same error

You have included the Google Maps API multiple times on this page. This may cause unexpected errors.

Mine is an SPA too, did you find a solution?

**Comment by jaimemoransantos** [comment]
> Is there a solution for the: `You have included the Google Maps API multiple times on this page. This may cause unexpected errors.` Error?

Hey @gegerino, just to let you know what seemed to be happening to me.

I have a Vue SPA built with Vue CLI, so on my main.js I wrote

```
appInstance.use(VueGoogleMaps, {
  load: {
    key: "my-key",
    libraries: "places,geocoding",
    v: "weekly",
  },
});
```

And then proceeded to use two components from the library in a single component

```
<GMapAutocomplete
            :value="autocomplete"
            placeholder="Dirección domiciliaria"
            @place_changed="getAddressData"
            required
            :readonly="isDisabled"
            class="form-control col-lg-12"
          >
          </GMapAutocomplete>
```

and

```
<GMapMap
            @click="mark"
            :center="center"
            :zoom="16"
            map-type-id="terrain"
            style="width: 100%; height: 250px"
            v-if="!loadingAutocomplete"
            :options="{
              zoomControl: true,
              mapTypeControl: true,
              scaleControl: true,
              streetViewControl: true,
              rotateControl: true,
              fullscreenControl: true,
              disableDefaultUi: true,
            }"
          >
            <GMapMarker
              :key="index"
              v-for="(m, index) in markers"
              :position="m.position"
              :clickable="true"
              :draggable="true"
              @click="center = m.position"
              @dragend="handleMarkerDrag"
            />
          </GMapMap>
```

So both loaded at the same time and consequently both of them called a method to load the Google Maps API script at the same time, causing the error/warning. So what I did is I loaded first component, the Autocomplete, while the GMap would conditionally render when a loading state was false and so I first requested with `navigator.geolocation` the latitude and longitude to get a marker and then defined my geocoder object `this.geocoder = new google.maps.Geocoder();` which could already be used because the Autocomplete component loaded the  Google Maps API script, with it I requested the address, and once the address loaded I changed the loadingAutocomplete to false and when the GMap rendered it worked automatically because the script was loaded previously. 

And the warning  disappeared.

TL;DR

If your case is the same as mine, that you have two components of the library rendered on the page, load one first and once it renders then you can render the other. Because when rendered simultaneously both call to load the Google Maps API script and thats when it can load more than once and cause errors. So doing it one by one allows the next component to use the Google Maps API script that is already loaded.
